webgpu.h

```diff
@@ -0,0 +1,1025 @@
+} wgpuPipelineStageDescriptor;
+
+typedef struct wgpuRasterizationStateDescriptor {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+    wgpuFrontFace frontFace;
+    wgpuCullMode cullMode;
+    int32_t depthBias;
+    float depthBiasSlopeScale;
+    float depthBiasClamp;
+} wgpuRasterizationStateDescriptor;
+
+typedef struct wgpuRenderPassDepthStencilAttachmentDescriptor {
+    wgpuTextureView attachment;
+    wgpuLoadOp depthLoadOp;
+    wgpuStoreOp depthStoreOp;
+    float clearDepth;
+    wgpuLoadOp stencilLoadOp;
+    wgpuStoreOp stencilStoreOp;
+    uint32_t clearStencil;
+} wgpuRenderPassDepthStencilAttachmentDescriptor;
+
+typedef struct wgpuSamplerDescriptor {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+    wgpuAddressMode addressModeU;
+    wgpuAddressMode addressModeV;
+    wgpuAddressMode addressModeW;
+    wgpuFilterMode magFilter;
+    wgpuFilterMode minFilter;
+    wgpuFilterMode mipmapFilter;
+    float lodMinClamp;
+    float lodMaxClamp;
+    wgpuCompareFunction compare;
+} wgpuSamplerDescriptor;
+
+typedef struct wgpuShaderModuleDescriptor {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+    uint32_t codeSize;
+    uint32_t const * code;
+} wgpuShaderModuleDescriptor;
+
+typedef struct wgpuStencilStateFaceDescriptor {
+    wgpuCompareFunction compare;
+    wgpuStencilOperation failOp;
+    wgpuStencilOperation depthFailOp;
+    wgpuStencilOperation passOp;
+} wgpuStencilStateFaceDescriptor;
+
+typedef struct wgpuTextureViewDescriptor {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+    wgpuTextureFormat format;
+    wgpuTextureViewDimension dimension;
+    uint32_t baseMipLevel;
+    uint32_t mipLevelCount;
+    uint32_t baseArrayLayer;
+    uint32_t arrayLayerCount;
+    wgpuTextureAspect aspect;
+} wgpuTextureViewDescriptor;
+
+typedef struct wgpuVertexAttributeDescriptor {
+    uint32_t shaderLocation;
+    uint64_t offset;
+    wgpuVertexFormat format;
+} wgpuVertexAttributeDescriptor;
+
+typedef struct wgpuBindGroupDescriptor {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+    wgpuBindGroupLayout layout;
+    uint32_t bindingCount;
+    wgpuBindGroupBinding const * bindings;
+} wgpuBindGroupDescriptor;
+
+typedef struct wgpuBindGroupLayoutDescriptor {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+    uint32_t bindingCount;
+    wgpuBindGroupLayoutBinding const * bindings;
+} wgpuBindGroupLayoutDescriptor;
+
+typedef struct wgpuColorStateDescriptor {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+    wgpuTextureFormat format;
+    wgpuBlendDescriptor alphaBlend;
+    wgpuBlendDescriptor colorBlend;
+    wgpuColorWriteMask writeMask;
+} wgpuColorStateDescriptor;
+
+typedef struct wgpuComputePipelineDescriptor {
+    const void* nextInChain;
+    wgpuStructureType structureType;
+    wgpuPipelineLayout layout;
+    wgpuPipelineStageDescriptor const * computeStage;
```

Comment:
why is this a pointer?

Reply:
Because this descriptor needs to be extended and could be swapped for a differently typed descriptor (linked to [1])